### PR TITLE
Improve menu experience and allow bitmap dimming

### PR DESCRIPTION
**Pull request by victorvuelma**
- Hour format menu: Change title to "Hour format", and when user changes, shows "12" or "24" format;
- Graphic change menu: Changed the behavior of the limits, when trying to advance beyond the displays, it returns to the initial one, when trying to return while being in the first one, it returns to the last one;
- Dimming: Allow bitmaps dimming using TFT_eSPI alphaBlend method;
- Clock: Adjusting time after change timezone;
- TFT: Improve display energy toggle, keeping screen black;
- Timezone: Improve TFT display update after a timezone update, mainly when dimming was toggle;
- Offset menu: Fix minutes offset display (was -1:-30, now it its -1:30);

### CODE REVIEW

**Comment by victorvuelma** [comment]
Hi @aly-fly,

Can you give your review about this PR?

**Comment by aly-fly** [comment]
Reviewed all changes. Looking good! Thanks for contributions.

File: EleksTubeHAX_pio/src/main.cpp [hunk 1]
```diff
@@ -295,16 +304,22 @@ void loop() {
         time_t offset = uclock.getTimeZoneOffset();
         int8_t offset_hour = offset/3600;
         int8_t offset_min = (offset%3600)/60;
+        if(offset_min < 0) {
+          offset_min = -offset_min;
+        }
```
Comment (SmittyHalibut) [inline EleksTubeHAX_pio/src/main.cpp:309]: Do you want the negative?  Or `+60` here?  When you're talking about modulo, I think `+60` is correct.

File: EleksTubeHAX_pio/src/main.cpp [hunk 2]
```diff
@@ -313,14 +328,20 @@ void loop() {
...
```
Comment (SmittyHalibut) [inline EleksTubeHAX_pio/src/main.cpp:333]: Same as modulo above.

**Review by SmittyHalibut — COMMENTED** [review]
I know this is already merged, but please see my reviews below.  If I'm correct and these warrant fixing, then submit another PR. (Or is there a way to update a PR that's already been merged?  I'm not that git-hard-core to know that off the top of my head.)

Thanks.

File: EleksTubeHAX_pio/src/Clock.h [hunk 3]
```diff
@@ -35,20 +35,31 @@ class Clock {
...
   int8_t getActiveGraphicIdx()          { return config->selected_graphic; }
   void adjustClockGraphicsIdx(int8_t adj) {
-    config->selected_graphic += adj;
-    if (config->selected_graphic > tfts.NumberOfClockFaces) { config->selected_graphic = tfts.NumberOfClockFaces; }
-    if (config->selected_graphic < 1) { config->selected_graphic = 1; }   
+    int8_t newGraphic = getActiveGraphicIdx();
+    newGraphic += adj;
+    
+    if (newGraphic > tfts.NumberOfClockFaces) { newGraphic = 1; }
+    if (newGraphic < 1) { newGraphic = tfts.NumberOfClockFaces; }
```
[the remark this answers is not in the document]
Comment (aly-fly) [reply EleksTubeHAX_pio/src/Clock.h:54]: it is 1 based. Data files are from 10 to 79.

File: EleksTubeHAX_pio/src/Clock.h [hunk 4]
```diff
@@ -35,20 +35,31 @@ class Clock {
   void toggleBlankHoursZero()           { config->blank_hours_zero = !config->blank_hours_zero; }
 
   // Internal time is kept in UTC. This affects the displayed time.
-  void setTimeZoneOffset(time_t offset) { config->time_zone_offset = offset; }
+  void setTimeZoneOffset(time_t offset) { 
+    config->time_zone_offset = offset; 
+    loop();
```
[the remark this answers is not in the document]
Comment (aly-fly) [reply EleksTubeHAX_pio/src/Clock.h:40]: Agree. It is a great way to make a runaway condition. Removed the "loop()".

File: EleksTubeHAX_pio/src/Clock.h [hunk 5]
```diff
@@ -35,20 +35,31 @@ class Clock {
...
+  }
   time_t getTimeZoneOffset()            { return config->time_zone_offset; }
-  void adjustTimeZoneOffset(time_t adj) { config->time_zone_offset += adj; }
+  void adjustTimeZoneOffset(time_t adj) { 
+    config->time_zone_offset += adj;
+    loop();
```
[the remark this answers is not in the document]
Comment (aly-fly) [reply EleksTubeHAX_pio/src/Clock.h:45]: Agree. It is a great way to make a runaway condition. Removed the "loop()".

File: EleksTubeHAX_pio/src/main.cpp (hunk 1 above)
Comment (aly-fly) [reply EleksTubeHAX_pio/src/main.cpp:309]: What is needed here is just removing the negative sign in front of the minutes.
If you want to show time offset of "-2 and 3/4" hours is this -2h and 45 min. Calculation will show you "-2:-45" and we want "-2:45". 
On the other hand -45 + 60 = 15 which is not a correct indication.

**Pull request merged**